src/main/java/com/rocketden/main/service/RoomService.java

```diff
@@ -137,7 +154,17 @@ public RoomDto removeUser(String roomId, RemoveUserRequest request) {
         }
 
         // Assign new host if user being kicked is host
-        conditionallyUpdateRoomHost(room, userToDelete);
+        if (room.getHost().equals(userToDelete)) {
+
+            // If the host is the last user, delete the room and return info.
+            if (room.getUsers().size() == 1) {
+                DeleteRoomRequest deleteRoomRequest = new DeleteRoomRequest();
+                deleteRoomRequest.setHost(request.getUserToDelete());
+                return deleteRoom(roomId, deleteRoomRequest);
+            }
+
+            return conditionallyUpdateRoomHost(room, userToDelete, true);
```

Comment:
Suggestion: instead of returning here, could we just remove the user and then call the conditional room host update afterwards? That way we could avoid passing an extra boolean flag to handle an edge case

Reply:
So, that's what was originally done, but there was a bug where two separate actions would be made on the database - the host would be changed to a regular user and a new host is instated, and the old host is deleted - these two database updates seemed to conflict as the host was not actually being deleted from the room. Therefore, I changed the function to accept a parameter so `repository.save(room)` is only called once, and no database conflicts occur (this seemed to fix the problem, so I assumed this was the issue).

Reply:
Ah I see thanks - looks like another concurrency issue similar to what was discovered in #122. I do think it'd be good to not have to rely on passing booleans across methods, but I guess since there's potentially no easy fix until the concurrency stuff is implemented, this looks good for this PR 👍